sql/core/src/main/scala/org/apache/spark/sql/jdbc/MySQLDialect.scala

```diff
@@ -94,4 +95,11 @@ private case object MySQLDialect extends JdbcDialect {
   override def getTableCommentQuery(table: String, comment: String): String = {
     s"ALTER TABLE $table COMMENT = '$comment'"
   }
+
+  override def getJDBCType(dt: DataType): Option[JdbcType] = dt match {
+    // See SPARK-35446: MySQL treats REAL as a synonym to DOUBLE by default
+    // We override getJDBCType so that FloatType is mapped to FLOAT instead
+    case FloatType => Option(JdbcType("FLOAT", java.sql.Types.FLOAT))
```

Comment:
@mariosmeim-db sorry one last thing. can we update migration guide too? https://github.com/apache/spark/blob/master/docs/sql-migration-guide.md